library/Icinga/Web/Navigation/ConfigMenu.php

```diff
@@ -0,0 +1,293 @@
+                ]
+            ]
+        ]
+    ];
+
+    protected $selected;
+
+    protected $cogItemActive = false;
+
+    protected function assembleUserMenuItem(HtmlElement $userMenuItem)
+    {
+        $username = Auth::getInstance()->getUser()->getUsername();
+
+        $userMenuItem->add(
+            HtmlElement::create(
+                'a',
+                Attributes::create(['href' => Url::fromPath('account')]),
+                [
+                    HtmlElement::create(
+                        'i',
+                        Attributes::create(['class' => 'user-ball']),
+                        Text::create($username[0])
+                    ),
+                    $username
+                ]
+            )
+        );
+
+        if (Icinga::app()->getRequest()->getUrl()->matches('account')) {
+            $userMenuItem->addAttributes([ 'class' => 'selected active']);
+        }
+    }
+
+    protected function assembleCogMenuItem($cogMenuItem)
+    {
+        $cogMenuItem->add([
+            HtmlElement::create(
+                'a',
+                [],
+                [
+                    new Icon('cog'),
+                    $this->createHealthBadge(),
+                ]
+            ),
+            $this->createLevel2Menu()
+        ]);
+
+        if ($this->cogItemActive) {
+            $cogMenuItem->addAttributes([ 'class' => 'active' ]);
+        }
+    }
+
+    protected function assembleLevel2Nav(BaseHtmlElement $level2Nav)
+    {
+        $navContent = HtmlElement::create('div', ['class' => 'flyout-content']);
+        foreach ($this->children as $c) {
+            if (isset($c['permission']) && ! Auth::getInstance()->hasPermission($c['permission'])) {
+                continue;
+            }
+
+            if (isset($c['title'])) {
+                $navContent->add(HtmlElement::create(
+                    'h3',
+                    [],
+                    t($c['title'])
+                ));
+            }
+
+            $ul = HtmlElement::create('ul', ['class' => 'nav']);
+            foreach ($c['items'] as $key => $item) {
+                $ul->add($this->createLevel2MenuItem($item, $key));
+            }
+            $navContent->add($ul);
+        }
+        $level2Nav->add($navContent);
+    }
+
+    protected function getHealthCount()
+    {
+        $count = 0;
+        $title = null;
+        $worstState = null;
+        foreach (HealthHook::collectHealthData()->select() as $result) {
+            if ($worstState === null || $result->state > $worstState) {
+                $worstState = $result->state;
+                $title = $result->message;
+                $count = 1;
+            } elseif ($worstState === $result->state) {
+                $count++;
+            }
+        }
+
+        switch ($worstState) {
+            case HealthHook::STATE_OK:
+                $count = 0;
+                break;
+            case HealthHook::STATE_WARNING:
+                $this->state = self::STATE_WARNING;
```

Comment:
This is defined dynamically. Initialize it properly please.

Reply:
✅